plugins/modules/azure_rm_afdorigin.py

```diff
@@ -0,0 +1,513 @@
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            weight=dict(type='int')
+        )
+
+        self.azure_origin = None
+        self.enabled_state = None
+        self.host_name = None
+        self.http_port = None
+        self.https_port = None
+        self.origin_host_header = None
+        self.priority = None
+        self.shared_private_link_resource = None
+        self.weight = None
+
+        self.name = None
+        self.origin_group_name = None
+        self.profile_name = None
+        self.resource_group = None
+        self.state = None
+
+        self.origin_client = None
+
+        self.results = dict(changed=False)
+
+        super(AzureRMOrigin, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            supports_check_mode=True,
+            supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+
+        self.origin_client = self.get_origin_client()
+
+        to_be_updated = False
+
+        response = self.get_origin()
+
+        if self.state == 'present':
+
+            if not response:
+                self.log("Need to create the Origin")
+
+                if not self.check_mode:
+                    new_response = self.create_origin()
+                    self.results['id'] = new_response['id']
+                    self.results['host_name'] = new_response['host_name']
+
+                self.results['changed'] = True
+
+            else:
+                self.log('Results : {0}'.format(response))
+                self.results['id'] = response['id']
+                self.results['host_name'] = response['host_name']
+
+                if response['host_name'] != self.host_name and self.host_name:
+                    to_be_updated = True
+                if response['http_port'] != self.http_port and self.http_port:
+                    to_be_updated = True
+                if response['https_port'] != self.https_port and self.https_port:
+                    to_be_updated = True
+                if response['origin_host_header'] != self.origin_host_header and self.origin_host_header:
+                    to_be_updated = True
+                if response['priority'] != self.priority and self.priority:
+                    to_be_updated = True
+                if response['weight'] != self.weight and self.weight:
+                    to_be_updated = True
+                if response['enabled_state'] != self.enabled_state and self.enabled_state:
+                    to_be_updated = True
+                if response['shared_private_link_resource']:
+                    if response['shared_private_link_resource']['group_id'] != \
+                            self.shared_private_link_resource.get('group_id') and \
```

Comment:
self.shared_private_link_resource, if not configured in the playbook, this parameter is None, so before comparing, please make sure that this value is not None, thank you!